File: backend/src/main/java/com/example/demo/enums/TaskStatus.java
package com.example.demo.enums;

public enum TaskStatus {
    PENDING("Pending"),
    TODO("Todo"),
    IN_PROGRESS("In Progress"),
    COMPLETED("Completed");

    private final String displayName;

    TaskStatus(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public TaskStatus getNextStatus() {
        return switch (this) {
            case PENDING -> TODO;
            case TODO -> IN_PROGRESS;
            case IN_PROGRESS -> COMPLETED;
            case COMPLETED -> COMPLETED;
        };
    }

    public boolean canTransitionTo(TaskStatus targetStatus) {
        return switch (this) {
            case PENDING -> targetStatus == TODO;
            case TODO -> targetStatus == IN_PROGRESS;
            case IN_PROGRESS -> targetStatus == COMPLETED;
            case COMPLETED -> false;
        };
    }
}

File: frontend/src/app/models/task-status.ts
export enum TaskStatus {
    PENDING = 'PENDING',
    TODO = 'TODO',
    IN_PROGRESS = 'IN_PROGRESS',
    COMPLETED = 'COMPLETED'
  }
  
  export const TaskStatusLabels: Record<TaskStatus, string> = {
    [TaskStatus.PENDING]: 'Pending',
    [TaskStatus.TODO]: 'Todo',
    [TaskStatus.IN_PROGRESS]: 'In Progress',
    [TaskStatus.COMPLETED]: 'Completed'
  };
  
  export function getNextStatus(currentStatus: TaskStatus): TaskStatus | null {
    switch (currentStatus) {
      case TaskStatus.PENDING:
        return TaskStatus.TODO;
      case TaskStatus.TODO:
        return TaskStatus.IN_PROGRESS;
      case TaskStatus.IN_PROGRESS:
        return TaskStatus.COMPLETED;
      case TaskStatus.COMPLETED:
        return null;
      default:
        return null;
    }
  }